### ISSUE TITLE
self.thre_loader与self.train_loader数据处理不同

### ISSUE THREAD

**Issue by wuyue5** [DAMO-DI-ML/KDD2023-DCdetector#26]
data_loader.py文件中，为什么train中使用窗口划分时，进行了高重叠，step为1，但使用的self.thre_loader是直接以win_size划分，没有重叠？这点比较疑惑，希望您能解答一下。
另外，想问一下这个模型可以实现实时的异常检测吗？

**Comment by yyysjz1997** [comment]
感谢您的疑问，"self.thre_loader是直接以win_size划分"是为了模型加速，使用step的话速度会慢一些些。如果追求效果的话，可以使用全部test/train dataset作为阈值的计算。使用test dataset作为阈值效果最好，train dataset的话我这边尝试过未经调参的，效果有些下降，可能的原因在于训练和测试集的数据分布不同。

对于实时异常检测，我们[最新的论文](https://arxiv.org/abs/2306.10347)中有对于每一种超参数推理时间和内存占用的讨论。如果追求实时异常检测，可以尝试一下另一个[DCdetector的复现版本](https://github.com/xuhongzuo/DeepOD/blob/main/deepod/models/time_series/dcdetector.py)，这个应该可以满足您说的实时异常检测。调用方法如下：

`
from deepod.models.time_series import DCdetector
clf = DCdetector()
clf.fit(X_train)
pred, scores = clf.decision_function(X_test)

from deepod.metrics import point_adjustment
from deepod.metrics import ts_metrics_enhanced
adj_eval_metrics = ts_metrics_enhanced(labels, point_adjustment(labels, scores), pred)
`

希望可以解决您的问题

**Comment by yyysjz1997** [comment]
> data_loader.py文件中，为什么train中使用窗口划分时，进行了高重叠，step为1，但使用的self.thre_loader是直接以win_size划分，没有重叠？这点比较疑惑，希望您能解答一下。 另外，想问一下这个模型可以实现实时的异常检测吗？

感谢您的疑问，"self.thre_loader是直接以win_size划分"是为了模型加速，使用step的话速度会慢一些些。如果追求效果的话，可以使用全部test/train dataset作为阈值的计算。使用test dataset作为阈值效果最好，train dataset的话我这边尝试过未经调参的，效果有些下降，可能的原因在于训练和测试集的数据分布不同。

对于实时异常检测，我们[最新的论文](https://arxiv.org/abs/2306.10347)中有对于每一种超参数推理时间和内存占用的讨论。如果追求实时异常检测，可以尝试一下另一个[DCdetector的复现版本](https://github.com/xuhongzuo/DeepOD/blob/main/deepod/models/time_series/dcdetector.py)，这个应该可以满足您说的实时异常检测。调用方法如下：

`from deepod.models.time_series import DCdetector
clf = DCdetector()
clf.fit(X_train)
pred, scores = clf.decision_function(X_test)
from deepod.metrics import point_adjustment
from deepod.metrics import ts_metrics_enhanced
adj_eval_metrics = ts_metrics_enhanced(labels, point_adjustment(labels, scores), pred)`

希望可以解决您的问题
